Bluetooth SCO USB driver

gowthambabu92 opened the pull request:
This driver is required if you want to use Bluetooth devices with USB interface to support SCO for BT HFP call.

Tests:
1. Flash Android
2. In adb shell, check #>cat proc/asound/cards
3. btaudiosource sound card should be listed

Tracked-On: OAM-126126

gowthambabu92 committed:
Bluetooth SCO USB driver

This driver is required if you want to use Bluetooth devices
with USB interface to support SCO for BT HFP call.

Tests:
1. Flash Android
2. In adb shell, check #>cat proc/asound/cards
3. btaudiosource sound card should be listed

Tracked-On: OAM-126126
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>

JeevakaPrabu commented on drivers/bluetooth/btintel.c:
```diff
@@ -3061,6 +3061,7 @@ static int btintel_setup_combined(struct hci_dev *hdev)
 	}
 
 	/* Apply the common HCI quirks for Intel device */
+        set_bit(HCI_QUIRK_RESET_ON_CLOSE, &hdev->quirks);
```
is this change specific to BT SCO or other issue?

gowthambabu92 replied:
This change fixes below issue.

Bluetooth: btusb: Add reset on close quirk for Intel adapters

Intel Bluetooth adapter is not exiting the loopback mode on bluetooth
socket closure. So, when bluetooth socket is opened again, bluetooth is
not working.

As per Bluetooth core specification, bluetooth adapter should exit loop
back mode on hci reset. Add HCI quirk reset to make sure adapter exits
loopback mode on bluetooth socket closure.

JeevakaPrabu committed:
Bluetooth: btusb: Add reset on close quirk for Intel adapters

Intel Bluetooth adapter is not exiting the loopback mode on bluetooth
socket closure. So, when bluetooth socket is opened again, bluetooth is
not working.

As per Bluetooth core specification, bluetooth adapter should exit loop
back mode on hci reset. Add HCI quirk reset to make sure adapter exits
loopback mode on bluetooth socket closure.

Tracked-On: OAM-126126
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Amrita Raju <amrita.raju@intel.com>

prashanth-intel committed:
Bluetooth: btusb: Add reset on close quirk for Intel adapters

Intel Bluetooth adapter is not exiting the loopback mode on bluetooth
socket closure. So, when bluetooth socket is opened again, bluetooth is
not working.

As per Bluetooth core specification, bluetooth adapter should exit loop
back mode on hci reset. Add HCI quirk reset to make sure adapter exits
loopback mode on bluetooth socket closure.

Tracked-On: OAM-126126
Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
SUCCESS: Android CI has completed Engineering Build for this issue.Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is SUCCESS. Please check the linked Tracked-On issue/Android CI Web  for more details. For Binaries: /cactus-absp-or-local/celadon-merge/789
